ppo/agents/vanilla_ppo.py

```diff
@@ -23,18 +31,67 @@ def __init__(
         key: chex.PRNGKey,
         learning_rate: float = 1e-3,
         discount: float = 0.99,
+        clipping_ratio_threshold: float = 0.1,
+        max_grad_norm: float = 0.5,
     ):
         super(VanillaPPO, self).__init__(observation_spec, policy_network, value_network, key, learning_rate, discount)
         self.replay_buffer = FixedReplayBuffer(buffer_capacity=50)
+        self.clipping_ratio_threshold = clipping_ratio_threshold
+        self.max_grad_norm = max_grad_norm
 
     def observe_first(self, timestep: dm_env.TimeStep) -> None:
         self.replay_buffer.add_first(timestep)
 
     def observe(self, value: float, log_probability: float, action: np.ndarray, next_timestep: dm_env.TimeStep) -> None:
         self.replay_buffer.add(value, log_probability, action, next_timestep)
 
-    def update(self):
-        pass
-
     def add_advantage(self, advantages):
         self.replay_buffer.add_advantage(advantages)
+
+    def value_loss(self, value_params, batch):
```

Comment:
Small detail, add typing in the method signature  `value_params: hk.Params, batch: Transition`